include/shad/core/unordered_map.h

```diff
@@ -50,6 +50,9 @@ template <class Key, class T, class Hash = shad::hash<Key>>
 class unordered_map {
   using hashmap_t = Hashmap<Key, T>;
 
+  template <typename T_>
+  friend class buffered_insert_iterator;
```

Comment:
This says that `buffered_insert_iterator<shad::array>` is a friend of `shad::unordered_map`.

This line should read as:
```C++
friend class buffered_insert_iterator<hashmap_t>;
```